djangosaml2idp/processors.py

```diff
@@ -1,21 +1,37 @@
+from django.conf import settings
+
+
 class BaseProcessor:
     """ Processor class is used to determine if a user has access to a client service of this IDP
         and to construct the identity dictionary which is sent to the SP
     """
 
+    def __init__(self, entity_id):
+        self._entity_id = entity_id
+
     def has_access(self, user):
         """ Check if this user is allowed to use this IDP
         """
         return True
 
+    def is_enabled(self, request):
+        """Is this saml2 integration enabled"""
+        return True
+
     def enable_multifactor(self, user):
         """ Check if this user should use a second authentication system
         """
         return False
 
-    def create_identity(self, user, sp_mapping):
+    def get_user_id(self, user):
+        user_field = getattr(settings, 'SAML_IDP_DJANGO_USERNAME_FIELD', None) or \
```

Comment:
Ok to move this ```getattr(settings, 'SAML_IDP_DJANGO_USERNAME_FIELD', None)``` part to top of the file so it only needs to retrieve the setting once instead of on every request?

Reply:
Yes, makes perfect sense. I'll get this done.

Reply:
I think there's arguably not much benefit to doing this.  getattr might be slow, but it is already being called twice in the same method.   Then because it is moved out of the method, it's not as straight forward to override it in the test, e.g: 

This no longer works:

```    def test_extract_user_id_configure_by_settings(self, settings):
        """Should use `settings.SAML_IDP_DJANGO_USERNAME_FIELD` to determine the user id field"""

        settings.SAML_IDP_DJANGO_USERNAME_FIELD = 'first_name'```

I'm not hugely precious either way, but I think it's better as it is currently. What do you think?

Reply:
That's a reasonable argument to keep it like it is for now. I think there might be some optimization possible in general (regarding the processor instantiation on every request) but that can be done at a later stage.